inception/decoder_tpl.go

```diff
@@ -399,7 +399,7 @@ mainparse:
 				{{range $byte, $fields := $si.FieldsByFirstByte}}
 				case '{{$byte}}':
 					{{range $index, $field := $fields}}
-						{{if ne $index 0 }}} else if {{else}}if {{end}} bytes.Equal(ffj_key_{{$si.Name}}_{{$field.Name}}, kn) {
+						{{if ne $index 0 }}} else if {{else}}if {{end}} len(kn) == {{len $field.JsonName}}-2 && bytes.Equal(ffj_key_{{$si.Name}}_{{$field.Name}}, kn) {
```

Comment:
bytes.Equal does this as its first check... so doing this is just because the compiler isnlt inlining `bytes.Equal`?